pkg/kubectl/scheme/sparse_test.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheme
+
+import (
+	"testing"
+
+	"k8s.io/api/batch/v1beta1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/diff"
+)
+
+func TestCronJob(t *testing.T) {
+	src := &v1beta1.CronJob{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}
+
+	encoder := Codecs.LegacyCodec(v1beta1.SchemeGroupVersion)
+	cronjobBytes, err := runtime.Encode(encoder, src)
```

Comment:
not sure this test is actually hitting the case that usually fails. encoding usually fails because we do something like this, where we construct the encoder with the list of all registered versions, not the version we want:
```
func InternalVersionJSONEncoder() runtime.Encoder {
	encoder := legacyscheme.Codecs.LegacyCodec(legacyscheme.Registry.RegisteredGroupVersions()...)
	return unstructured.JSONFallbackEncoder{Encoder: encoder}
}
```

Reply:
See the other test

Reply:
Added it anyway, but the other test covered it already.